### PR TITLE
Specify line ending and formatting rules

### PR DESCRIPTION
**Pull request by Xeeynamo**
I need to gain again some momentum in this project. This is a very small PR that suggests contributors to strict code formatting rules. I also became aware that shell scripts on Linux do not work with `crlf` line ending. Since there are minor inconsistencies between the text encoding and line ending, the rules will force the editor to always use UTF-8 without BOM and `lf` in all occasions with few exceptions.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,56 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+indent_style = space
+indent_size = space
```
Comment (kenjiuno) [inline .editorconfig:8]: Perhaps this will be `indent_size = 4`
According to https://editorconfig.org/

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+tab_size = 4
```
Comment (kenjiuno) [inline .editorconfig:9]: `tab_size = 4` → `indent_size = 4`

**Review by kenjiuno — COMMENTED** [review]
Thanks, please check comments about `.editorconfig`

I also recommend to run `dotnet-format --workspace OpenKh.sln` once.

```bat
dotnet tool install -g dotnet-format
dotnet-format --workspace OpenKh.sln
```

This will enforce your changes immediately, and then for incoming pull requests, it will reduce extra diffs coming from white space changes.

File: .editorconfig (hunk 1 above)
Comment (Xeeynamo) [reply .editorconfig]: Sorry, I did not paid attention to it. Fixed 👍

File: .editorconfig (hunk 2 above)
Comment (Xeeynamo) [reply .editorconfig:9]: I suggest to keep `tab_size` for the code that already use tabulations rather than spaces

**Comment by Xeeynamo** [comment]
> I also recommend to run `dotnet-format --workspace OpenKh.sln` once.

I was not aware of that tool. Thank you! Although the problem is that I get a lot of files changed. I prefer to not push it as it would give big problems when merging pull requests.

**Review by kenjiuno — APPROVED** [review]

**Pull request merged**